bodhi/server/consumers/automatic_updates.py

```diff
@@ -0,0 +1,158 @@
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""
+The Bodhi handler that creates updates automatically from tagged builds.
+
+This module is responsible for the process of creating updates when builds are
+tagged with certain tags.
+"""
+
+import logging
+import re
+
+import fedora_messaging
+
+from bodhi.server import buildsys, initialize_db
+from bodhi.server.config import config
+from bodhi.server.exceptions import BodhiException
+from bodhi.server.models import Build, ContentType, Package, Release
+from bodhi.server.models import Update, UpdateType, User
+from bodhi.server.util import transactional_session_maker
+
+
+log = logging.getLogger('bodhi')
+
+
+class AutomaticUpdateHandler:
+    """
+    The Bodhi Automatic Update Handler.
+
+    A consumer that listens for messages about tagged builds and creates
+    updates from them.
+    """
+
+    _auto_update_cfg_key_re = re.compile(
+        r'(?P<releasename>\w+)\.autoupdates\.(?P<subkey>(?:\w+\.)*\w+)', re.ASCII)
+
+    _version_number_re = re.compile(r'[^\d]*(?P<vernum>\d+).*', re.ASCII)
+
+    def __init__(self, db_factory: transactional_session_maker = None):
+        """
+        Initialize the Automatic Update Handler.
+
+        Args:
+            db_factory: If given, used as the db_factory for this handler. If
+            None (the default), a new TransactionalSessionMaker is created and
+            used.
+        """
+        if not db_factory:
+            initialize_db(config)
+            self.db_factory = transactional_session_maker()
+        else:
+            self.db_factory = db_factory
+
+    def __call__(self, message: fedora_messaging.api.Message) -> None:
+        """Create updates from appropriately tagged builds.
+
+        Args:
+            message: The message we are processing.
+        """
+        msg = message.body.get('msg')
+
+        if not msg:
+            log.debug("Ignoring message without 'msg'.")
+            return
+
+        missing = []
+        for mandatory in ('tag', 'build_id', 'name', 'version', 'release'):
+            if mandatory not in msg:
+                missing.append(mandatory)
+        if missing:
+            raise BodhiException(f"Received incomplete tag message. Missing: {', '.join(missing)}")
+
+        btag = msg['tag']
+        bname = msg['name']
+        bversion = msg['version']
+        brelease = msg['release']
+        bnvr = f'{bname}-{bversion}-{brelease}'
```

Comment:
I don't have a strong opinion here. but maybe we don't need the `bname`, `bversion` and `brelease` variable seems they are used only in the f-string to create `bnvr`.  I guess the following would just be as readable
```suggestion
        bnvr = f"msg['name']-msg['version']-msg['release']"
```

Reply:
Yeah, I think I only used it as a shortcut. I'll probably use old-style `str.format()` instead because it's less to type. :wink: